spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/inbound/ReactiveMongoDbMessageSource.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.inbound;
+
+import org.reactivestreams.Publisher;
+
+import org.springframework.data.mongodb.ReactiveMongoDatabaseFactory;
+import org.springframework.data.mongodb.core.ReactiveMongoOperations;
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import org.springframework.data.mongodb.core.query.BasicQuery;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.expression.Expression;
+import org.springframework.expression.TypeLocator;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.expression.spel.support.StandardTypeLocator;
+import org.springframework.integration.endpoint.AbstractMessageSource;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.mongodb.support.MongoHeaders;
+import org.springframework.integration.support.AbstractIntegrationMessageBuilder;
+import org.springframework.util.Assert;
+
+import com.mongodb.DBObject;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * @author David Turanski
+ *
+ * @since 5.3
+ */
+public class ReactiveMongoDbMessageSource extends AbstractMessageSource<Publisher<?>> {
+	private final Expression queryExpression;
+
+	private volatile Expression collectionNameExpression = new LiteralExpression("data");
+
+	private volatile StandardEvaluationContext evaluationContext;
+
+	private volatile ReactiveMongoOperations reactiveMongoTemplate;
+
+	private volatile MongoConverter mongoConverter;
+
+	private volatile ReactiveMongoDatabaseFactory reactiveMongoDatabaseFactory;
+
+	private volatile boolean initialized = false;
```

Comment:
Copied from the non reactive version.  Also passes checkstyle

Reply:
No one modified non-reactive `MessageSource` for a while, so that's why that `volatile` is still there...
There is just no Checkstyle rule for ordering properties